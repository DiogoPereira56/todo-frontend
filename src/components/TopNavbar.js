import '../fonts.css'
import {Container, Navbar, Nav, Ul, Li, Img, Center, Input} from './TopNavbar.styles'
import iconSquared from '../imgs/iconSquared.png'
import settings from '../imgs/settings.png'
import help from '../imgs/help.png'
import whatsNew from '../imgs/whatsNew.png'

const TopNavbar = () => {
  return (
    <Container>
        <Navbar>
            <Img src={iconSquared} alt="" />
            <b>To Do</b>
            <Center><Input placeholder="Search"></Input></Center>
            <Nav>
                <Ul>
                    <Li><Img src={settings} alt="" /></Li>
                    <Li><Img src={help} alt="" /></Li>
                    <Li><Img src={whatsNew} alt="" /></Li>
                    <Li><p>Diogo Henrique</p></Li>
                </Ul>
            </Nav>
        </Navbar>
    </Container>
  );
}

export default TopNavbar;