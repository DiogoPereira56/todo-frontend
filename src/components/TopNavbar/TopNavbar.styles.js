import styled from 'styled-components'

const Container = styled.div`
    margin: auto;
    padding: 10px;
    background: #0e005e;
`

const Navbar = styled.div`
    display: flex;
    align-items: center;
    color: #ffffff;
`

const Nav = styled.nav`
    text-align: right;
`
const Ul = styled.ul`
    display: inline-block;
    list-style-type: none;
    vertical-align: center;
`
const Li = styled.li`
    display: inline-block;
    margin-right: 20px;
`

const Img = styled.img`
    width: 50px;
    padding-right:20px;
`

const Center = styled.div`
    flex: 1;
    text-align: center;
`
const Input = styled.input`
    padding: 8px;
    cursor: pointer;
    width: 400px;
    height: 32px;
    transition: background-color 0.1s;
    background: rgb(191, 207, 230);
    border: none;
    border-radius: 2px;
`

const A = styled.a`
    text-decoration: none;
    color: white;
`
export {Container, Navbar, Nav, Ul, Li, Img, Center, Input, A};