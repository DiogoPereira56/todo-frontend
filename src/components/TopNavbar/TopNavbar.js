import '../../fonts.css'
import {Container, Navbar, Nav, Ul, Li, Img, Center, Input, A} from './TopNavbar.styles'
import iconSquared from '../../imgs/iconSquared.png'
import settings from '../../imgs/settings.png'
import help from '../../imgs/help.png'
import whatsNew from '../../imgs/whatsNew.png'
import { useLazyQuery } from '@apollo/client'
import { LOG_OUT } from '../../graphQL/Queries'
import { PropTypes } from 'prop-types'
/* import {Warnning} from '../Login/LoginForm.style'
import {Options, Actions} from '../CenterColumn/CenterColumn.styles' */
//import { useState } from 'react'

const TopNavbar = ({name}) => {

  const [logOut, {loading}] = useLazyQuery(LOG_OUT);
  //const [showOptions, setShowOptions] = useState(false);

  return (
    <Container>
        <Navbar>
            <Img src={iconSquared} alt="" />
            <b>To Do</b>
            <Center><Input placeholder="Search"></Input></Center>
            <Nav>
                <Ul>
                    <Li><Img src={settings} alt="" /></Li>
                    <Li><Img src={help} alt="" /></Li>
                    <Li><Img src={whatsNew} alt="" /></Li>
                    <Li onClick={() => setShowOptions(!showOptions)}> <p>{name}</p> </Li>
                    <Li><p>| <A onClick={logOut} href="/">Log Out</A></p></Li>
                    {/* showOptions && (
                      <Options>
                          <Actions><Warnning>
                            <a onClick={logOut} href="/">Log Out</a>
                            </Warnning></Actions>
                      </Options>
                    ) */}
                </Ul>
            </Nav>
        </Navbar>
    </Container>
  );
}

TopNavbar.propTypes = {
  name: PropTypes.string
};

export default TopNavbar;