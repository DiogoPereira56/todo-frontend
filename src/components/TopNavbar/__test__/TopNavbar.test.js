/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import TopNavbar from '../TopNavbar';
import ReactDOM, { unmountComponentAtNode } from 'react-dom';
import { act, fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

//afterEach(cleanup); //done automatically after version 9.0.0

/* let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
}); */

describe('<TopNavbar />', () => {
    it('renders TopNavbar without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<TopNavbar></TopNavbar>, div);
    });

    it('renders TopNavbar correctly', () => {
        const { getByText } = render(<TopNavbar name="Diogo"></TopNavbar>);
        expect(getByText('Diogo'));
    });

    it('matches topnavbar snapshot', () => {
        const tree = renderer.create(<TopNavbar name="Diogo"></TopNavbar>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('changes the input values correctly', () => {
        const { debug, getByPlaceholderText } = render(<TopNavbar name="Diogo"></TopNavbar>);
        let searchInput = getByPlaceholderText('search');
        fireEvent.change(searchInput, { target: { value: 'task' } });
        searchInput = getByPlaceholderText('search');
        //console.log(searchInput);
        expect(searchInput).toHaveValue('task');
    });

    /* it('submits the form correctly', async () => {
        const handleSubmit = jest.fn();
        const { debug, getByPlaceholderText, getByTestId } = render(
            <TopNavbar name="Diogo" onSubmit={handleSubmit}></TopNavbar>,
        );
        debug();
        const searchInput = getByPlaceholderText('search');
        fireEvent.change(searchInput, { target: { value: 'task' } });
        const searchForm = getByTestId('submit');
        //fireEvent.keyDown(searchInput, { key: 'Enter', keyCode: 13 });
        //console.log(searchForm);
        fireEvent.input(searchForm);
        expect(handleSubmit).toHaveBeenCalled();
    }); */

    /* it('changes the input values correctly', () => {
        act(() => {
            render(<TopNavbar name="Diogo"></TopNavbar>, container);
        });
        //console.log(container.querySelector("[data-testid='search']"));
        let searchInput = container.querySelector("[data-testid='search']");
        fireEvent.change(searchInput, { target: { value: 'task' } });
        searchInput = container.querySelector("[data-testid='search']");
        expect(searchInput).toHaveValue('task');
    }); */

    /* it('submits the form correctly', () => {
        //jest.spyOn();
        const wrapper = shallow(
            <TopNavbar name="Diogo" handleSearchedTasks={handleSearchedTasks}></TopNavbar>,
        );
        let searchFormik = wrapper.find('Formik').first();
        console.log(searchFormik.debug());
        const teste = wrapper.find('Formik').simulate('submit', { preventDefault: () => {} });
        expect(teste).toHaveBeenCalled();
    }); */
});
