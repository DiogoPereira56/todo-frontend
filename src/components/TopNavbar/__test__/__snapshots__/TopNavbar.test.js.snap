// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNavbar /> matches topnavbar snapshot 1`] = `
<div
  className="sc-bdfBwQ kIYrtp"
>
  <div
    className="sc-gsTCUz ZQrBb"
  >
    <img
      alt=""
      className="sc-jSgupP kVoUAA"
      src="iconSquared.png"
    />
    <b>
      To Do
    </b>
    <div
      className="sc-gKsewC kijxMI"
    >
      <form
        action="#"
        data-testid="submit"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="sc-iBPRYJ gPwcxt"
          data-testid="search"
          name="search"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="search"
          value=""
        />
        <br />
      </form>
    </div>
    <nav
      className="sc-dlfnbm wAeD"
    >
      <ul
        className="sc-hKgILt fNjwa-d"
      >
        <li
          className="sc-eCssSg jzTusa"
        >
          <img
            alt=""
            className="sc-jSgupP kVoUAA"
            src="settings.png"
          />
        </li>
        <li
          className="sc-eCssSg jzTusa"
        >
          <img
            alt=""
            className="sc-jSgupP kVoUAA"
            src="help.png"
          />
        </li>
        <li
          className="sc-eCssSg jzTusa"
        >
          <img
            alt=""
            className="sc-jSgupP kVoUAA"
            src="whatsNew.png"
          />
        </li>
        <li
          className="sc-eCssSg jzTusa"
        >
          <p
            data-testid="name"
          >
            Diogo
          </p>
        </li>
        <li
          className="sc-eCssSg jzTusa"
        >
          |
        </li>
        <li
          className="sc-eCssSg jzTusa"
        >
          <p>
            <a
              className="sc-fubCfw gbeUsT"
              href="/"
              onClick={[Function]}
            >
              Log Out
            </a>
          </p>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
